test/integration/job/job_test.go

```diff
@@ -270,6 +270,158 @@ func TestIndexedJob(t *testing.T) {
 	validateJobSucceeded(ctx, t, clientSet, jobObj)
 }
 
+func TestSuspendJob(t *testing.T) {
+	testCases := []struct {
+		featureGate   bool
+		suspendCreate bool
+		suspendUpdate bool
+		shortSubtest  bool
+	}{
+		// Exhaustively test all combinations other than trivial true->true and
+		// false->false cases.
+		{featureGate: true, suspendCreate: false, suspendUpdate: true, shortSubtest: true},
+		{featureGate: true, suspendCreate: true, suspendUpdate: false},
+		{featureGate: false, suspendCreate: false, suspendUpdate: true},
+		{featureGate: false, suspendCreate: true, suspendUpdate: false},
+	}
+
+	for _, tc := range testCases {
+		name := fmt.Sprintf("featureGate=%v,suspendCreate=%v,suspendUpdate=%v", tc.featureGate, tc.suspendCreate, tc.suspendUpdate)
+		t.Run(name, func(t *testing.T) {
+			if testing.Short() && !tc.shortSubtest {
```

Comment:
I believe we should be able to run all these tests in parallel as they all run in different namespaces.

Reply:
would parallel testing work with feature gate testing? it _seems_ to work fine locally but this is the sort of the thing that breaks in CIs but works locally :)

I've removed the testing.Short() check for a t.Parallel(). let's see how the CI reacts.

Reply:
ahhh, the gate is probably a problem. An alternative would be to have hierarchical `t.Run` where the first level is not parallel (containing a single feature gate value). But not sure if it's worth the complexity.

Reply:
the CI seems to be happy with parallel testing + feature gates

Reply:
It seems error prone.. the gate values are global after all.

Reply:
yeah... I think the CI is passing because it's running with GOMAXPROCS=1 or something? anyway, went back to testing.Short() so that we run only one test in presubmits.